client/src/components/files/explorer/FileListDisplay.vue

```diff
@@ -153,6 +190,21 @@ async function selectAll(selected: boolean): Promise<void> {
   props.folders.selectAll(selected);
 }
 
+async function onHeaderSortChange(property: SortProperty): Promise<void> {
+  let newSortOrder = true;
```

Comment:
Shouldn't default to `true`, the order should remain the same when switching from one category to another.

You can simply set it as: 
```
const newSortOrder = props.currentSortProperty === property ? !props.currentSortOrder : props.currentSortOrder;
```

and remove the lines 196 to 198.